src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -194,9 +194,10 @@ edm4eic::Cluster CalorimeterClusterRecoCoG::reconstruct(const edm4eic::ProtoClus
   }
   if (tw == 0.) {
     m_log->warn("zero total weights encountered, you may want to adjust your weighting parameter.");
+  } else {
+    cl.setPosition(v / tw);
+    cl.setPositionError({}); // @TODO: Covariance matrix
```

Comment:
Should we fill the gaps with NaN values? Position {0,0,0} would not be correct, right?

Reply:
I didn't want to introduce any changes here, which is why I kept it at zero which is what the code currently does (setPosition fails, excecution continues without setting position).

I think the correct thing here is to not have a cluster. I would prefer not to write NaN since then all code downstream has to prepare for accepting NaNs, and NaNs infect everything in accumulated operations like averages.

Reply:
What actually happens here is that all hits are so low in energy that they don't rise above the specified logWeightBase value.  I think that would qualify as "there was no cluster after all".

Reply:
We'd probably have to change the function to return `std::optional<Cluster>`.

Reply:
See #1000